### ISSUE TITLE
Vulkan can't play RGB24 PNG files

### ISSUE THREAD

**Issue by CogentRedTester** [mpv-player/mpv#7981]
### Important Information

Provide following Information:
```
mpv 0.32.0-613-gb238897614 Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
 built on Sun Aug  2 08:33:54 +08 2020
FFmpeg library versions:
   libavutil       56.57.100
   libavcodec      58.98.100
   libavformat     58.49.100
   libswscale      5.8.100
   libavfilter     7.87.100
   libswresample   3.8.100
FFmpeg version: git-2020-08-01-15ef16dce
```
Windows 10 Version 2004
OS Build: 19041.388

Graphics Card: NVIDIA GT 1030
Driver Version: 451.67

Screenshots:

![image](https://user-images.githubusercontent.com/57207260/89618600-9bad2d80-d8cb-11ea-8967-c2dd76d001ae.png)

![image](https://user-images.githubusercontent.com/57207260/89618807-fb0b3d80-d8cb-11ea-968f-cd7a635d8b6d.png)

### Reproduction steps

Find an image with 24bit RGB colour encoding, try running it with Vulkan. It's happened with all the ones I've tried so far, though I don't have access to many.

### Expected behavior
The image will be displayed normally.

### Actual behavior
The image is simply a blue screen, as shown above, and the following errors are printed to the terminal:

```
[vo/gpu/vulkan/libplacebo] Validation failed: !params.uniform || params.size <= gpu->limits.max_ubo_size (../libplacebo/src/gpu.c:736)
[vo/gpu/vulkan/libplacebo] Failed creating buffer for tex upload fallback!
[vo/gpu/vulkan/libplacebo] Validation failed: !params.uniform || params.size <= gpu->limits.max_ubo_size (../libplacebo/src/gpu.c:736)
[vo/gpu/vulkan/libplacebo] Failed creating buffer for tex upload fallback!
```

### Log file
[log2.txt](https://github.com/mpv-player/mpv/files/5039862/log2.txt)

### Sample files
[file1](https://user-images.githubusercontent.com/57207260/89619282-dbc0e000-d8cc-11ea-9715-bea508d77c5f.png)
[file2](https://user-images.githubusercontent.com/57207260/89619225-c51a8900-d8cc-11ea-9ece-52521db1884e.png)

**Comment by haasn** [comment]
Possible solutions:

1. Try an SSBO if UBO fails (greater size limit on your GPU, apparently)
2. Use strided/partial uploads (annoying but doable)

**Comment by snylonue** [comment]
Same issue on windows with GTX1050 and my mpv version is `0.33.0-56-g3e175dff4a`
I got the same error message

**Comment by haasn** [comment]
Moved [upstream](https://code.videolan.org/videolan/libplacebo/-/issues/125)

**Comment by haasn** [comment]
Fixed in [libplacebo@89fc2b41](https://code.videolan.org/videolan/libplacebo/-/commit/89fc2b4117c884f72c3357317d8b0e1cec856e6e)
